dev-client/src/components/common/ListFilter.tsx

```diff
@@ -0,0 +1,278 @@
+import {Button, FormControl, HStack, Input, Select, VStack} from 'native-base';
+import {useCallback, useMemo, useRef, useState} from 'react';
+import {
+  Modal,
+  ModalMethods,
+} from 'terraso-mobile-client/components/common/Modal';
+import EmbadgedIcon from 'terraso-mobile-client/components/common/EmbadgedIcon';
+import {Icon, IconButton} from 'terraso-mobile-client/components/common/Icons';
+
+type Getter<Item> = keyof Item | ((item: Item) => string);
+
+type InputFilterConfig<Item> = {
+  key: Getter<Item>;
+};
+
+type SelectFilterConfig<Item, S> = {
+  key: keyof Item;
+  lookup?: Record<string, S[keyof S]>;
+};
+
+export type FilterConfig<Item, S> = {
+  textInput: InputFilterConfig<Item>;
+  select?: Record<keyof S, SelectFilterConfig<Item, S>>;
+};
+
+export type OptionMapping<T> = {[Property in keyof T]: string};
+
+type FilterModalProps<SelectIDs extends OptionMapping<SelectIDs>> = {
+  selectFilters?: SelectFilterDisplayConfig<SelectIDs>;
+  selectFilterUpdate: (id: keyof SelectIDs) => (newValue: string) => void;
+  filterValues: Record<keyof SelectIDs, string | null>;
+  applyFilter: () => void;
+};
+
+type FilterDisplayConfig = {
+  label: string;
+  placeholder: string;
+};
+
+type TextInputFilterDisplayConfig = FilterDisplayConfig;
+
+type SelectFilterDisplayConfig<T extends OptionMapping<T>> = {
+  [Property in keyof T]: {
+    options: Record<T[Property], string>;
+  } & FilterDisplayConfig;
+};
+
+export type ListFilterProps<
+  Item,
+  SelectIDs extends {[Property in keyof SelectIDs]: string} = {},
+> = {
+  items: Item[];
+  filterConfig: FilterConfig<Item, SelectIDs>;
+  displayConfig: {
+    textInput: TextInputFilterDisplayConfig;
+    select?: SelectFilterDisplayConfig<SelectIDs>;
+  };
+  children: (value: {
+    filteredItems: Item[];
+    InputFilter: React.ReactNode;
+  }) => React.ReactNode;
+};
+
+const normalizeText = (text: string) =>
+  text
+    .trim()
+    .toLowerCase()
+    .normalize('NFD')
+    .replace(/[\u0300-\u036f]/g, ''); // unicode range for combining diacritical marks
```

Comment:
issue: this should be a global helper function, not specific to this component. Also: we do something similar in web client. See if that can be reused.